### PR TITLE
applications: serial_lte_modem Unsolicited result code (URC) format

### PR DESCRIPTION
**Pull request by junqingzou**
To help AT client to parse URC event message:
.Use comma ',' to separate each component in URC
.Use double quotes '"' to wrap string type component
.Add `<CR><LF>` before OK, ERROR, +CME, +CMS command result

DevZone ticket: 261674

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 1]
```diff
@@ -1037,7 +1037,7 @@ Response syntax
 
 ::
 
-   #XTCPSVR: <handle> started
+   #XTCPSVR: <handle>, started
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1041]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 2]
```diff
@@ -1048,7 +1048,7 @@ Unsolicited notification
 
 ::
 
-   #XTCPSVR: <error> stopped
+   #XTCPSVR: <error>, stopped
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1052]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 3]
```diff
@@ -1188,14 +1188,14 @@ Response syntax
 
 ::
 
-   #XTCPCLI: <handle> connected
+   #XTCPCLI: <handle>, connected
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1192]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 4]
```diff
@@ -1188,14 +1188,14 @@ Response syntax
...
 
 Unsolicited notification
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 ::
 
-   #XTCPCLI: <error> disconnected
+   #XTCPCLI: <error>, disconnected
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1199]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 5]
```diff
@@ -1416,7 +1416,7 @@ Response syntax
 
 ::
 
-   #XUDPSVR: <handle> started
+   #XUDPSVR: <handle>, started
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1420]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 6]
```diff
@@ -1427,7 +1427,7 @@ Unsolicited notification
 
 ::
 
-   #XUDPSVR: <error> stopped
+   #XUDPSVR: <error>, stopped
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1431]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 7]
```diff
@@ -1555,14 +1555,14 @@ Response syntax
 
 ::
 
-   #XUDPCLI: <handle> connected
+   #XUDPCLI: <handle>, connected
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1559]: Missed double quote for string.

File: applications/serial_lte_modem/doc/TCPIP_AT_commands.rst [hunk 8]
```diff
@@ -1555,14 +1555,14 @@ Response syntax
...
 
 Unsolicited notification
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 ::
 
-   #XUDPCLI: <error> disconnected
+   #XUDPCLI: <error>, disconnected
```
Comment (lats1980) [inline applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:1566]: Missed double quote for string.

File: applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c [hunk 9]
```diff
@@ -673,14 +673,15 @@ static int handle_at_mqtt_connect(enum at_cmd_type cmd_type)
 	case AT_CMD_TYPE_READ_COMMAND:
 		if (ctx.sec_transport) {
 			sprintf(rsp_buf,
-				"#XMQTTCON: %d,\"%s\",\"%s\",\"%s\","
-				"\"%s\",%d,%d\r\n",
+				"#XMQTTCON: %d, \"%s\", \"%s\", \"%s\","
+				"\"%s\", %d, %d\r\n",
 				ctx.connected, ctx.cid, ctx.uname,
```
Comment (lats1980) [inline applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c:678]: A missed space before \"%s\"

				" \"%s\", %d, %d\r\n",

File: applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c [hunk 10]
```diff
@@ -673,14 +673,15 @@ static int handle_at_mqtt_connect(enum at_cmd_type cmd_type)
...
 				ctx.pword, ctx.url, ctx.port,
 				ctx.sec_tag);
 		} else {
-			sprintf(rsp_buf, "#XMQTTCON: %d,\"%s\",\"%s\",\"%s\""
-				",\"%s\",%d\r\n",
+			sprintf(rsp_buf,
+				"#XMQTTCON: %d, \"%s\", \"%s\", \"%s\""
+				",\"%s\", %d\r\n",
 				ctx.connected, ctx.cid, ctx.uname, ctx.pword,
```
Comment (lats1980) [inline applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c:685]: A missed space before "%s"

			" \"%s\", %d, %d\r\n",

**Review by lats1980 — COMMENTED** [review]
Some missing space or double quotes as comment in line.

**Review by rlubos — APPROVED** [review]

**Review by lats1980 — APPROVED** [review]
Looks good!

**Review by ru-fu — CHANGES_REQUESTED** [review]
The current doc changes look good, but there are also a lot of AT commands and responses in slm_description.rst - these must probably be updated as well.

**Comment by junqingzou** [comment]
> slm_description.rst

Added, please review

**Review by ru-fu — APPROVED** [review]
I don't know if these are all changes that are required or if some are missing (for example, there is an XACCEPT response in line 1069 that might need an update as well?).
But the changes that are made look good. :)

**Pull request merged**